src/FileStorage.php

```diff
@@ -49,28 +47,26 @@
     private $options;
 
     /**
-     * @var MessageCatalogue[] Fetched catalogies
+     * @var MessageCatalogue[] Fetched catalogues
      */
     private $catalogues;
 
     /**
-     * @param TranslationWriter                                            $writer
-     * @param SymfonyTranslationLoader|TranslationLoader|TranslationReader $loader
-     * @param array                                                        $dir
-     * @param array                                                        $options
+     * @param TranslationWriterInterface $writer
+     * @param TranslationReaderInterface $reader
+     * @param array                      $dir
+     * @param array                      $options
      */
-    public function __construct(TranslationWriter $writer, $loader, array $dir, array $options = [])
+    public function __construct($writer, $reader, array $dir, array $options = [])
```

Comment:
I think we can still keep `TranslationWriter` typehint for `$writer`, it's the same class where just a method was renamed.

Reply:
Yeah, but we want people to be able to provide their own class. 

If the BC layer was removed I would type hint this to TranslationWriterInterface.